src/GitHub.UI/Behaviours/AddEmptyItemToList.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Interactivity;
+
+namespace GitHub.UI
+{
+    public class AddEmptyItemToList : Behavior<ListBox>
+    {
+        object previousSelection = null;
+        dynamic defaultValue;
+        protected override void OnAttached()
+        {
+            base.OnAttached();
+
+            AssociatedObject.SelectionChanged += (s, e) =>
+            {
+                if (e.AddedItems.Count > 0)
+                {
+                    if (previousSelection == null)
+                    {
+                        previousSelection = AssociatedObject.SelectedItem;
+                        if (defaultValue == null)
+                        {
+                            var binding = AssociatedObject.GetBindingExpression(ListBox.ItemsSourceProperty);
+                            if (binding == null)
+                            {
+                                var m = BindingOperations.GetMultiBindingExpression(AssociatedObject, ListBox.ItemsSourceProperty);
+                                binding = (BindingExpression)m.BindingExpressions[0];
```

Comment:
It wasn't clear to me that `m` is guaranteed not to be null here. Maybe we should assert that.